provisioning/baremetal_pod.go

```diff
@@ -263,15 +263,20 @@ func createInitContainerIpaDownloader(images *Images) corev1.Container {
 	return initContainer
 }
 
-func createInitContainerMachineOsDownloader(info *ProvisioningInfo) corev1.Container {
-	optionValue := ""
+func ipOptionForMachineOsDownloader(info *ProvisioningInfo) string {
+	var optionValue string
 	switch info.NetworkStack {
 	case NetworkStackV4:
 		optionValue = "ip=dhcp"
 	case NetworkStackV6:
 		optionValue = "ip=dhcp6"
+	case NetworkStackDual:
+		optionValue = ""
 	}
+	return optionValue
+}
 
+func createInitContainerMachineOsDownloader(info *ProvisioningInfo) corev1.Container {
```

Comment:
This function is used by both the os-downloader in the metal3 pod and the one in the image cache, so we end up applying the command line change twice before the image reaches the host.
This _probably_ doesn't break anything, since it works by doing `s/^options/options ip=dhcp/` (in perl) but it's not ideal.

Reply:
Actually, I think it might break something: the checksum will be different depending on which host the API VIP is on. There might be a scenario in which the image and checksum get read either side of an API VIP failover, in which case they won't match.
I don't know that we test without a provisioning network in CI so this never gets covered.
I opened #163 to fix this.